sapp/ui/frontend/src/Filter.js

```diff
@@ -956,6 +957,13 @@ const Filter = (props: {refetch: any, refetching: boolean}) => {
     window.sessionStorage.setItem('filter', JSON.stringify(currentFilter));
   }, [currentFilter]);
 
+  const clearFilter = () => {
+    setCurrentFilter(emptyFilter);
+    // Hacky solution to the hacky clearAndRefetch function
+    props.refetch(filterToVariables(emptyFilter));
```

Comment:
It seems like we set `currentFilter` to `emptyFilter` then effectively call `props.refetch(filterToVariables(emptyFilter))` twice. What's the reason for calling `props.refetch` twice here? Is it because the UI doesn't update after the first `props.refetch` call?

Reply:
ya it doesn't, the clearAndRefetch function is done in such a way to reload only when new data is available which a single refetch call isn't being able to do!